packages/valory/skills/transaction_settlement_abci/tests/test_behaviours.py

```diff
@@ -1068,6 +1179,47 @@ def test_check_tx_history_behaviour(
         )
 
 
+class TestCheckLateTxHashesBehaviour(TransactionSettlementFSMBehaviourBaseCase):
+    """Test CheckLateTxHashesBehaviour."""
+
+    def _fast_forward(self, late_arriving_tx_hashes: str) -> None:
+        """Fast-forward to relevant behaviour."""
+
+        agent_address = self.skill.skill_context.agent_address
+        kwargs = dict(
+            safe_contract_address="safe_contract_address",
+            participants=frozenset({agent_address, "a_1", "a_2"}),
+            participant_to_signature={},
+            most_voted_tx_hash="",
+            late_arriving_tx_hashes=late_arriving_tx_hashes,
+        )
+        abci_app_db = AbciAppDB(setup_data=AbciAppDB.data_to_lists(kwargs))
+
+        self.fast_forward_to_behaviour(
+            behaviour=self.behaviour,
+            behaviour_id=CheckLateTxHashesBehaviour.auto_behaviour_id(),
+            synchronized_data=TransactionSettlementSynchronizedSata(abci_app_db),
+        )
+
+        current_behaviour = self.behaviour.current_behaviour
+        current_behaviour_id = cast(BaseBehaviour, current_behaviour).behaviour_id
+        assert current_behaviour_id == CheckLateTxHashesBehaviour.auto_behaviour_id()
+
+    def test_check_tx_history_behaviour(self) -> None:
+        """Test CheckTransactionHistoryBehaviour."""
```

Comment:
test for history on `CheckLateTxHashesBehaviour`

https://github.com/valory-xyz/open-autonomy/blob/4efaa677ebc268e0a2448e6496e2649ee684e51a/packages/valory/skills/transaction_settlement_abci/behaviours.py#L622-L633